src/common/checksum.c

```diff
@@ -577,6 +667,17 @@ dcb_is_valid(const daos_csum_buf_t *csum)
 	       csum->cs_nr > 0;
 }
 
+void
+dcb_insert(daos_csum_buf_t *dcb, int idx, uint8_t *csum_buf, size_t len)
```

Comment:
this is really "append/add/attach" not insert

Reply:
Insert makes sense to me. It takes an index so the csum_buf can be copied anywhere into the daos_csum_buf_t csums.